factory/factory.go

```diff
@@ -56,3 +53,17 @@ func CheckConfigVersion() error {
 
 	return nil
 }
+
+func validateWebuiUri(uri string) error {
+	parsedUrl, err := url.ParseRequestURI(uri)
+	if err != nil {
+		return err
+	}
+	if parsedUrl.Scheme != "http" && parsedUrl.Scheme != "https" {
+		return fmt.Errorf("unsupported scheme for webuiUri: %s", parsedUrl.Scheme)
+	}
+	if parsedUrl.Host == "" {
```

Comment:
```suggestion
	if parsedUrl.Hostname() == "" {
```
and you could add a UTs for `http://:8080" as invalid URI